Interactable Changes

Karkaranos opened the pull request:
- Objects will give one of each intent per emotion
- Objects can be interacted with once
- Controls for customizing questions and responses

New interactable canvas visuals
![image](https://github.com/user-attachments/assets/780835c3-7294-4a3d-849a-665ca078268b)

Editor with new options 
![image](https://github.com/user-attachments/assets/40c56a02-8380-4351-aa59-5ccb17fa7225)

Karkaranos committed:
Interactable Changes

- Objects will give one of each intent per emotion
- Objects can be interacted with once
- Controls for customizing questions and responses

TheBeesSneeze commented on Nonstick Summer 2025/Assets/Scripts/Player Interaction/OpenCanvasInteractable.cs:
```diff
@@ -1,5 +1,16 @@
+using TMPro;
+using System;
 
 public class OpenCanvasInteractable : MonoBehaviour, IInteractable
```
the idea was to put all the quiz logic into the canvas that gets opened, rather than putting it on here, so that the opencanvasinteractable script couId be generalized.
i don't think it matters that much atp, esp because this code is good. 
onIy do this if you have to make other changes- do you think you could rename this script to be a bit more specific?

Karkaranos replied:
Yeah I can rename it for that

TheBeesSneeze commented on Nonstick Summer 2025/Assets/Scripts/Player Interaction/OpenCanvasInteractable.cs:
```diff
@@ -12,30 +23,89 @@ public class OpenCanvasInteractable : MonoBehaviour, IInteractable
+    /// <summary>
+    /// If giving one card
+    /// </summary>
+    [Obsolete("This function is no longer needed")]
```
maybe just deIete this function or put it in a #region

Karkaranos replied:
Sounds good

TheBeesSneeze commented on Nonstick Summer 2025/Assets/Scripts/Player Interaction/OpenCanvasInteractable.cs:
```diff
@@ -51,3 +121,21 @@ private void OnDrawGizmos()
+Date Created :          June 19, 2025
+Date Modified :         June 19, 2025
+Brief Description :     Stores information for interactable object questions
+                        Toby, if you would rather I switch this to a scriptable object later, I can
```
i Iove avoiding making scriptable objects whenever possibIe. as Iong as it stiII Iets design do whatever they want, i have no qualms !

Karkaranos committed:
Small Tweaks

- Renamed OpenCanvasInteractable
- Removed Obselete function
- You can now control Interact canvas text in the inspector
